cmsController.php

```diff
@@ -0,0 +1,195 @@
+ * @param $db to add to
+ */
+function updatePortfolio (array $postData,PDO $db) {
+    if(array_key_exists('submitPf',$postData)) {
+        $query = $db->prepare("REPLACE INTO `portfolioItems`(`title`,`description`,`imgRef`,`projURL`,`github`)
+                                     VALUES (:title, :descr, :picRef, :url, :github);");
+        $query->bindValue(':title', $postData['pfTitle']);
+        $query->bindValue(':descr', $postData['pfDesc']);
+        $query->bindValue(':url', $postData['pfURL']);
+        $query->bindValue(':github', $postData['githubURL']);
+        $query->bindValue(':picRef', $postData['picSelect']);
+        $query->execute();
+    }
+}
+
+/**
+ * Takes list of Image names from DB and returns an array
+ *
+ * @param $db PDO to select from
+ * @return array of images, values
+ */
+function getImgDropDown (PDO $db):array {
+    $query=$db->prepare("SELECT `id`,`name` FROM `images` WHERE `deleted` !=1;");
+    $query->execute();
+    $items=$query->fetchall();
+    return $items;
+}
+
+/**
+ * Takes array of images and creates a string of HTML options for each image
+ *
+ * @param $arr Array of images, values
+ *
+ * @return string of HTML options for images
+ */
+function makeImgDropDown (array $arr):string {
+    $resultString = "";
+    foreach ($arr as $item) {
+        $resultString .= '<option value="' . $item['id'] . '">' . $item['name'] . '</option>';
+    }
+    return $resultString;
+}
+
+/**
+ * Updates the DB with deleted flag for selected portfolio item
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to amend
+ */
+function deletePfItem (array $postData,PDO $db) {
+    if(array_key_exists('pfDelete',$postData)) {
+        $item = $postData['pfDelete'];
+        $query = $db->prepare("UPDATE `portfolioItems` SET `deleted`=1 WHERE `title`=:item;");
+        $query->bindParam(':item',$item);
+        $query->execute();
+    }
+}
+
+/** Updates/adds Article based on name
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function UpdateArticle (array $postData,PDO $db) {
+    if(array_key_exists('artTitle',$postData)) {
+        $query = $db->prepare("REPLACE INTO `articles`(`title`,`description`,`url`)
+                              VALUES(:title, :descr, :url)");
+        $query->bindParam(':title', $postData['artTitle']);
+        $query->bindParam(':descr', $postData['artDesc']);
+        $query->bindParam(':url', $postData['artURL']);
+        $query->execute();
+    }
+}
+
+/**
+ * Updates the DB with a deleted flag for selected Article item
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to amend
+ */
+function DeleteArticle (array $postData,PDO $db) {
+    if(array_key_exists('artDelete',$postData)) {
```

Comment:
can we move this if (and the others in the functions) outside the function, into where the functions are called. atm you call EVERY function which is very slow

Reply:
I've applied this by moving all php function calls into a new file (cmsLogic.php) and using the IF statement to call the function, rather than the other way around.